extensions-core/s3-extensions/src/main/java/org/apache/druid/storage/s3/ServerSideEncryptingAmazonS3.java

```diff
@@ -204,7 +205,19 @@ public ServerSideEncryptingAmazonS3 build()
         throw new ISE("S3StorageConfig cannot be null!");
       }
 
-      return new ServerSideEncryptingAmazonS3(amazonS3ClientBuilder.build(), s3StorageConfig.getServerSideEncryption());
+      AmazonS3 amazonS3Client;
+      try {
+        amazonS3Client = RetryUtils.retry(
+            () -> amazonS3ClientBuilder.build(),
+            (e) -> true,
+            RetryUtils.DEFAULT_MAX_TRIES
+        );
+      }
+      catch (Exception e) {
+        throw new RuntimeException(e);
+      }
+
+      return new ServerSideEncryptingAmazonS3(amazonS3Client, s3StorageConfig.getServerSideEncryption());
```

Comment:
Had some questions here regarding `(e) -> true` part. I noticed that we use `(e) -> true` predicate at a couple other places. So I wanted to check if this is okay, as there can be multiple other implementation approaches here.

Possible approaches:
1. Keep it as is.
2. Create another predicate like `S3Utils.S3RETRY`, with the condition `e.getMessage().contains("Unable to find a region via the region provider chain")`, and use that. Reason for a different predicate is that S3RETRY's javadoc suggest that it's supposed to be used for I/O operations only.
3. Update S3RETRY predicate itself to handle `Unable to find a region via the region provider chain` message, and update javadoc to mark it non-I/O specific.

cc: @cryptoe @kfaraz Thoughts?

Reply:
Yeah, I don't think we want to retry unconditionally on all exceptions. If an error is non-recoverable, we should fail fast instead of retrying. 

Also, could you elaborate on why we need this exception to be retriable - "Unable to find a region via the region provider chain"?

Reply:
@abhishekrb19 

> Also, could you elaborate on why we need this exception to be retriable - "Unable to find a region via the region provider chain"?

There seem to be times where AWS SDK isn't able to obtain the credentials for using some AWS service. (https://github.com/aws/aws-sdk-java/issues/2285#issuecomment-613119333 and https://medium.com/expedia-group-tech/service-slow-to-retrieve-aws-credentials-ebc02a38e95b).

Also, in my chat with Dhananjay and Andy, they mentioned that retries solve this type of problem. Hence, the suggestion was to add retries since there's no way to deterministically simulate it.

> Yeah, I don't think we want to retry unconditionally on all exceptions. If an error is non-recoverable, we should fail fast instead of retrying.

Fair enough. Would you suggest going with approach (2)? Or approach (3)?

Reply:
Thanks for the clarification! 
IMO, it would make sense to do 3 and update `S3RETRY`'s javadoc as it seems stale since we handle non-IO exceptions in the retry code

Reply:
@abhishekrb19 Have made the change.